sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosOperations.java

```diff
@@ -123,6 +125,17 @@ public interface CosmosOperations {
      */
     <T> T insert(T objectToSave, PartitionKey partitionKey);
 
+    /**
+     * patches item
+     * @param entityToPatch must not be {@literal null}, must contain {@link org.springframework.data.annotation.Id}
+     * and {@link com.azure.spring.data.cosmos.core.mapping.PartitionKey} field
+     * @param patchOperations must not be {@literal null}, max operations is 10
+     * @param options Optional CosmosPatchItemRequestOptions, e.g. options.setFilterPredicate("FROM products p WHERE p.used = false");
+     * @param <T> type class of domain type
+     * @return the patched item
+     */
+    <T> T patch(T entityToPatch, CosmosPatchOperations patchOperations, CosmosPatchItemRequestOptions options);
```

Comment:
should we also introduce an API without the requestOptions? `<T> T patch(T entityToPatch, CosmosPatchOperations patchOperations);`